public/apps/login/login-page.tsx

```diff
@@ -220,9 +220,19 @@ export function LoginPage(props: LoginPageDeps) {
         }
         case AuthType.SAML: {
           const samlConfig = props.config.ui[AuthType.SAML].login;
-          formBodyOp.push(
-            renderLoginButton(AuthType.SAML, SAML_AUTH_LOGIN_WITH_FRAGMENT, samlConfig)
-          );
+          const urlParams = new URLSearchParams(window.location.search);
```

Comment:
Could this new logic be broken out into a function, I think that could help clarify while this block is doing and keep its logic isolated

Reply:
I abstracted this out into a function in login-utils